src/browser/ui/__tests__/ReactDOMComponent-test.js

```diff
@@ -140,6 +140,51 @@ describe('ReactDOMComponent', function() {
       expect(stubStyle.color).toEqual('green');
     });
 
+    it('should reject haxors on initial markup', function() {
+      var container = document.createElement('div');
+
+      expect(function() {
+        var element = React.createElement(
+          'x-foo-component',
+          {'blah" onclick="beevil" noise="hi': 'selected'},
+          null
+        );
+        React.render(element, container);
+      }).toThrow();
+    });
+
+    it('should reject haxors on update', function() {
+      var container = document.createElement('div');
+
+      var beforeUpdate = React.createElement('x-foo-component', {}, null);
+      React.render(beforeUpdate, container);
+
+      expect(function() {
+        var afterUpdate = React.createElement(
+          'x-foo-component',
+          {'blah" onclick="beevil" noise="hi': 'selected'},
+          null
+        );
+        React.render(afterUpdate, container);
+      }).toThrow();
```

Comment:
Ah, I was looking at the wrong test block.  And it's all starting to come back to me.  We throw the Invariant Violation on initial render and updateAttributeByID throws a `Invalid character: attribute name: blah" onclick="beevil" noise="hi` on update.  The error is different because the first is a check we do ourselves during initial render, the second is a check done by the browser (so the browser owns the error message).  We could do the check on update also, but that would mean we are doing the check twice (once ourselves and once by the browser), which is a waste of valuable cpu cycles (performance issue).  We can't expect a particular error message to be thrown because the specifics of the error message are browser-dependent and we want our unit tests to run in any browser.